improver/utilities/field_texture.py

```diff
@@ -0,0 +1,234 @@
+        # store them in a cube.
+        actual_transitions = potential_transitions.copy(
+            data=self._calculate_transitions(cube.data)
+        )
+
+        # Sum the number of actual transitions within the neighbourhood.
+        actual_transitions = SquareNeighbourhood(sum_or_fraction="sum").run(
+            actual_transitions, radius=radius
+        )
+        actual_transitions.data = np.where(cube.data > 0, actual_transitions.data, 0)
+
+        # Calculate the ratio of actual to potential transitions.
+        ratio = np.ones_like(actual_transitions.data)
+        np.divide(
+            actual_transitions.data,
+            potential_transitions.data,
+            out=ratio,
+            where=(cube.data > 0),
+        )
+
+        # Create a new cube to contain the resulting ratio data.
+        ratio = create_new_diagnostic_cube(
+            "texture_of_{}".format(self.cube_name),
+            1,
+            cube,
+            MANDATORY_ATTRIBUTE_DEFAULTS,
+            data=ratio,
+        )
+        return ratio
+
+    @staticmethod
+    def _calculate_transitions(data):
+        """
+        Identifies actual transitions present in a binary field. These transitions
+        are defined as the number of cells of value zero that directly neighbour
+        cells of value 1. The number of transitions is calculated for all cells
+        of value 1. The number of transitions for cells of value 0 is set to 0.
+
+        Args:
+            data (numpy.ndarray):
+                A NumPy array of the input cube for data manipulation.
+
+        Returns:
+            numpy.ndarray:
+                A NumPy array containing the transitions for ratio calculation.
+        """
+        padded_data = np.pad(data, 1, mode="edge")
+        diff_x = np.abs(np.diff(padded_data, axis=1))
+        diff_y = np.abs(np.diff(padded_data, axis=0))
+        cell_sum_x = diff_x[:, 0:-1] + diff_x[:, 1:]
+        cell_sum_y = diff_y[0:-1, :] + diff_y[1:, :]
+        cell_sum = cell_sum_x[1:-1, :] + cell_sum_y[:, 1:-1]
+        cell_sum = np.where(data > 0, cell_sum, 0)
+        return cell_sum
+
+    def process(self, cube):
+        """
+        Calculates a field of texture to use in differentiating solid and
+        more scattered features.
+
+        Args:
+            cube (iris.cube.Cube):
+                Input data in cube format containing the field for which the
+                texture is to be assessed.
+
+        Returns:
+            iris.cube.Cube:
+                A cube containing the mean of the thresholded ratios in cube
+                format.
+        """
+
+        values = np.unique(cube.data)
+        non_binary = np.where((values != 0) & (values != 1), True, False)
+        if non_binary.any():
+            raise ValueError("Incorrect input. Cube should hold binary data only")
+
+        # Create new cube name for _calculate_ratio method.
+        self.cube_name = cube.coord(var_name="threshold").name()
+
```

Comment:
As you pointed out, this fails if the cube doesn't have a probability coordinate.  Suggest instead using the utility from `improver.metadata.probabilistic` as follows:
```
self.cube_name = find_threshold_coordinate(cube).name()
```
This raises a useful error if the threshold coordinate is not found.

Reply:
Done.